src/progressbar/progressbar.ts

```diff
@@ -0,0 +1,36 @@
+import {Component, Input, ElementRef} from 'angular2/angular2';
+
+@Component({
+  selector: 'ngb-progressbar',
+  template: `
+    <progress class="progress {{extraClasses}}  {{type && 'progress-' + type}}" [value]="value" [max]="max">
+      <div class="progress">
+        <span class="progress-bar" [style.width.%]="calculatePorcentage()"><ng-content></ng-content></span>
+      </div>
+    </progress>
+  `
+})
+export class NgbProgressbar {
+  _value: number;
+  min = 0;
+  extraClasses: string;
+
+  @Input()
+  set value(newValue: number) {
+    this._value = newValue;
+    this.calculatePorcentage();
+  }
+
+  get value() { return this._value; }
+
+  @Input() type: string;
+
+  @Input() max = 100;
+
+  constructor(private element: ElementRef) {
```

Comment:
That would be fantastic. I'm not at home now, so I will check that later
El 5/11/2015 3:50 p. m., "Pawel Kozlowski" <notifications@github.com>
escribió:

> In src/progressbar/progressbar.ts
> <https://github.com/ng-bootstrap/core/pull/79#discussion_r44019244>:
>
> > +  min = 0;
> > +  extraClasses: string;
> > +
> > +  @Input()
> > +  set value(newValue: number) {
> > +    this._value = newValue;
> > +    this.calculatePorcentage();
> > +  }
> > +
> > +  get value() { return this._value; }
> > +
> > +  @Input() type: string;
> > +
> > +  @Input() max = 100;
> > +
> > +  constructor(private element: ElementRef) {
>
> We can do everything without injecting ElementRef:
>
>    - you can get classes set on an element by doing a setter for class,
>    see:
>    https://github.com/angular/angular/blob/master/modules/angular2/src/core/directives/ng_class.ts
>    - this will also make sure that the class is not propagated to the
>    host element so you won't need to clear it by hand
>
> In short: I don't think we need / should be injecting ElementRef here
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ng-bootstrap/core/pull/79/files#r44019244>.
>